### DISCUSSION TITLE
Why does the notification keep appearing for more than 5 seconds?

### DISCUSSION THREAD

**Discussion by nozomi-terasawa** [yuriykulikov/AlarmClock#694] [Q&A]
Hello @yuriykulikov.
I am creating a clock app based on your implementation.
 Thank you for the wonderful open-source clock app.

 I have one question. Why do the notifications during an active alarm continue to display for more than five seconds? With the system's default settings, pop-up notifications likely appear for about five seconds. However, in your app, they persist for longer, remaining visible until the user takes an action to stop or snooze, or swipes them away. I would like to know the secret behind this behavior.
 
 There probably isn't a setting in NotificationCompat.Builder to keep notifications persistent, so there must be an implementation by the developers to ensure they remain visible. I couldn't find that part in the source code, so I'm reaching out directly to inquire about it.

Thank you.

**Comment by nozomi-terasawa** [comment]
'm sorry, I found the solution on my own. By correctly implementing setFullScreenIntent, I was able to have it displayed continuously.

**Comment by yuriykulikov** [comment]
Hello,
Yes, it is a combination of notification channel settings and full screen
intent.  I am glad that it is working as expected now!
